harmonica/forward/tesseroid_layer.py

```diff
@@ -69,6 +69,8 @@ def tesseroid_layer(
     if properties:
         data_names = tuple(p for p in properties.keys())
         data = tuple(np.asarray(p) for p in properties.values())
+    # Check if tesseroid boundaries are overlapped
+    _check_overlap(coordinates[0])
```

Comment:
I think we should be running this after `_check_regular_grid`. When getting the spacing we are assuming that the longitude array is actually regular.